### DISCUSSION TITLE
Problem with Veins and Simu5G Installation on OMNeT++

### DISCUSSION THREAD

**Discussion by rimsayegh** [Unipisa/Simu5G#234] [Q&A]
Hi everyone,

I'm encountering an issue during the installation of Veins with Simu5G in OMNeT++. Here are the details of my setup:
Simu5G version: 1.2.2
INET version: 4.5
Veins version: 5.2
OMNeT++ version: 6.0.1
Operating System: Windows 10

I followed the instructions provided in the Simu5G documentation. However, I received the following error:

ld.lld: error: too many errors emitted, stopping now (use /errorlimit:0 to see all errors)
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [Makefile:121: ../out/clang-debug/src/libveins_inet_dbg.dll] Error 1
make[1]: Leaving directory '/c/omnetpp/omnetpp-6.0.1/veins_inet/src'
make: *** [Makefile:36: all] Error 2
"make MODE=debug -j12 all" terminated with exit code 2. Build might be incomplete.

It seems to be a linker issue. Could this be related to the INET version being too new for the current versions of Veins or Simu5G?
If so, which versions should I use, or are there any workarounds to resolve this issue?

Thank you for your help!

**Comment by rhornig** [comment]
I believe that Veins was never tested with Simu5G on Windows, so it will not work. I'm saying this because looking into the codebase of Simu5G there is absolutely no trace of windows DLL export macros on functions and classes. DLL export is absolutely necessary if you want to USE Simu5G from other libraries (Veins in this case).

You should go with a Linux virtual machine. WSL2 would work just fine and you can also try `opp_env` which will install those components automatically.
